src/chocolatey/infrastructure.app/builders/ConfigurationBuilder.cs

```diff
@@ -510,7 +510,7 @@ private static void SetEnvironmentOptions(ChocolateyConfiguration config)
 
         private static void SetLicensedOptions(ChocolateyConfiguration config, ChocolateyLicense license, ConfigFileSettings configFileSettings)
         {
-            config.Information.IsLicensedVersion = license.IsLicensedVersion();
+            config.Information.IsLicensedVersion = license.IsLicensedVersion() && license.AssemblyLoaded;

```

Comment:
This is wrong.

Even if you are unable to load the extension, or if it isn't present, as long as you have a licensed file it is considered to be a licensed version of Chocolatey CLI.

Additionally, by setting `IsLicensedVersion` to false will prevent the installation of `chocolatey.extension` to happen.

For what you brought up internally, you need to update the `EnvironmentSettings` class that I linked to you to.

![image](https://github.com/chocolatey/choco/assets/1474648/74f86e64-a34a-463c-9b31-78b8c7faf5f4)
![image](https://github.com/chocolatey/choco/assets/1474648/0e3d455e-2908-4554-bfa9-03b265ad7126)

Reply:
Ah, that would explain the failure installing the licensed extension 😅 . I erroneously thought the error was related to the strongname mismatch that I had forced.

My reasoning for trying to do it here was that we don't have a License object at the EnvironmentSettings, then thinking of it, it seemed like we might have similar checks that might fail elsewhere. So this seemed like it would solve the issue, as well as stave off other potential issues. Clearly that's not the case. I'll get this part backed out and add the check where we initially discussed (and do a quick check of other places we have similar checks).

Reply:
Actually, this error occurs even with the check on the line mentioned in Slack. It looks like the `ChocolateyLicenseType` environment variable is set immediately after the updated check in EnvironmentSettings.

Even funner, the change to get the license causes all sorts of more output that is probably undesirable (Each time we run `License.ValidateLicense()` we output if the license file is there but assembly is not).

<img width="1024" alt="2023-06-26_14-33-46" src="https://github.com/chocolatey/choco/assets/30301021/7a315fea-5907-4fa1-ae67-0b575309f7d8">

I've made some more changes, just testing them out locally before pushing up to GitHub, then will kick off some Test Kitchen runs both locally and in Team City.